### PR TITLE
Remove Python bindings from Makefiles

### PR DESCRIPTION
**Pull request by alexreinking**
This is in service of easing ongoing maintenance after #6815, which will add a `setup.py` script for hooking into the Python build system. It goes through CMake since there is robust and well maintained support for that (via `scikit-build`).

I might have to write CMake builds for `apps/onnx` and `apps/HelloPyTorch` now (this should be done at some point, regardless), but we'll see what the buildbots have to say.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1982,20 +1975,18 @@ TEST_APPS=\
 TEST_APPS_DEPS=$(TEST_APPS:%=%_test_app)
 BUILD_APPS_DEPS=$(TEST_APPS:%=%_build_app)
 
-$(BUILD_APPS_DEPS): distrib build_python_bindings
```
Comment (steven-johnson) [inline Makefile:1985]: I think some of the apps may require the python bindings... maybe onnx? Not sure

File: python_bindings/readme.md [hunk 2]
```diff
@@ -57,18 +58,15 @@ The best way to get set up is to use a virtual environment:
 ```console
 $ python3 -m venv venv
 $ . venv/bin/activate
-$ pip install -r requirements.txt 
+$ python3 -m pip install -U setuptools wheel
```
Comment (steven-johnson) [inline python_bindings/readme.md:61]: Drive-by change, I presume, but what is this about?

File: Makefile (hunk 1 above)
Comment (alexreinking) [reply Makefile:1985]: Yeah, I said in the PR description that apps/onnx and apps/HelloPyTorch might break. Was going to wait and see.

File: python_bindings/readme.md (hunk 2 above)
Comment (alexreinking) [reply python_bindings/readme.md:61]: Regarding `pip` versus `python3 -m pip`, this prevents issues with overwriting open files on Windows while upgrading, see here: https://stackoverflow.com/a/60786507/2137996

**Comment by alexreinking** [comment]
> (Also, a followup here may be necessary to stop the buildbots from testing python on Make builds)

Yeah, we might need to add a dummy `test_python` target that does nothing to get this through the buildbots.

**Comment by alexreinking** [comment]
> I might have to write CMake builds for `apps/onnx` and `apps/HelloPyTorch` now (this should be done at some point, regardless), but we'll see what the buildbots have to say.

Looks like these _aren't_ affected by this change! Or perhaps not tested...?

**Comment by steven-johnson** [comment]
Windows failures are not flakes:

https://buildbot.halide-lang.org/master/#/builders/45/builds/322/steps/10/logs/stdio

```
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\Error.h(16): warning C4275: non dll-interface class 'std::runtime_error' used as base for dll-interface struct 'Halide::Error'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\include\stdexcept(101): note: see declaration of 'std::runtime_error'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\Error.h(16): note: see declaration of 'Halide::Error'
```

...not sure why this is happening now but not before?

**Comment by alexreinking** [comment]
There is _absolutely_ no way this PR causes that error, right? The Windows build is fully through CMake and the only CMake "change" here is that I added a comment somewhere I found confusing.

It looks like somehow `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` failed on Win64 but not Win32?

**Comment by alexreinking** [comment]
Ahhhh @steven-johnson that's not the error, _this_ is the error:

```
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2065: 'ASanGlobalsMetadataAnalysis': undeclared identifier
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2923: 'llvm::RequireAnalysisPass': 'ASanGlobalsMetadataAnalysis' is not a valid template type argument for parameter 'AnalysisT'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): note: see declaration of 'ASanGlobalsMetadataAnalysis'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2976: 'llvm::RequireAnalysisPass': too few template arguments
C:\src\build_bot\worker\llvm-15-x86-64-windows\llvm-install\include\llvm/IR/PassManager.h(1250): note: see declaration of 'llvm::RequireAnalysisPass'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2641: cannot deduce template arguments for 'llvm::RequireAnalysisPass'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2783: 'llvm::RequireAnalysisPass<AnalysisT,IRUnitT,AnalysisManagerT,ExtraArgTs...> llvm::RequireAnalysisPass(void)': could not deduce template argument for 'AnalysisT'
C:\src\build_bot\worker\llvm-15-x86-64-windows\llvm-install\include\llvm/IR/PassManager.h(1250): note: see declaration of 'llvm::RequireAnalysisPass'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2783: 'llvm::RequireAnalysisPass<AnalysisT,IRUnitT,AnalysisManagerT,ExtraArgTs...> llvm::RequireAnalysisPass(void)': could not deduce template argument for 'IRUnitT'
C:\src\build_bot\worker\llvm-15-x86-64-windows\llvm-install\include\llvm/IR/PassManager.h(1250): note: see declaration of 'llvm::RequireAnalysisPass'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2780: 'llvm::RequireAnalysisPass<AnalysisT,IRUnitT,AnalysisManagerT,ExtraArgTs...> llvm::RequireAnalysisPass(llvm::RequireAnalysisPass<AnalysisT,IRUnitT,AnalysisManagerT,ExtraArgTs...>)': expects 1 arguments - 0 provided
C:\src\build_bot\worker\llvm-15-x86-64-windows\llvm-install\include\llvm/IR/PassManager.h(1250): note: see declaration of 'llvm::RequireAnalysisPass'
C:\src\build_bot\worker\halide-testbranch-main-llvm15-x86-64-windows-cmake\halide-source\src\CodeGen_LLVM.cpp(1172): error C2672: 'llvm::PassManager<llvm::Module,llvm::AnalysisManager<IRUnitT>>::addPass': no matching overloaded function found
```

**Comment by abadams** [comment]
What's the build-time difference between the Makefile and the newer path? IIRC, I originally resurrected the Makefile because it built the python bindings 10x faster than setup.py, because setup.py had problems with parallel builds.

**Comment by alexreinking** [comment]
The new setup.py in #6815 will shell out to CMake/Ninja for building the native code, so parallelism is not a concern.

It's also going to play nicer with existing Python build stuff, so it's not quite apples-to-apples. For instance, I don't think the Makefiles produced distributable wheels or source tarballs, which is what PyPI expects.
